rust/index/src/hnsw_provider.rs

```diff
@@ -438,7 +438,9 @@ impl HnswIndexProvider {
             match self.remove_temporary_files(index_id).await {
                 Ok(_) => {}
                 Err(e) => {
-                    tracing::error!("Failed to remove temporary files: {}", e);
+                    if e.kind() != std::io::ErrorKind::NotFound {
```

Comment:
total nit - prefer a match here stylistically 

what are the valid cases for this?

Reply:
By valid cases are you asking if we should ever be responding to such an error?

Reply:
Yes I am saying why does this error happen?

Reply:
This is the case we talked about regarding the cache not being in sync with the files on disk.  Our short-term end state was having the cache evict the files and taking the race condition because we will eventually need a file cleaner anyway.  Avoiding this race condition requires much undesirable code.